bindings/rust/s2n-tls/src/testing/resumption.rs

```diff
@@ -0,0 +1,335 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        connection,
+        session_ticket::{SessionTicket, SessionTicketCallback},
+        testing::{s2n_tls::*, *},
+    };
+    use std::{cell::RefCell, rc::Rc, time::Duration};
+
+    // Creates session ticket callback handler
+    #[derive(Default, Clone)]
+    pub struct SessionTicketRawBytesHandler {
+        stored_ticket: Rc<RefCell<Vec<u8>>>,
+    }
+
+    // Implement the session ticket callback
+    impl SessionTicketCallback for SessionTicketRawBytesHandler {
+        fn on_session_ticket(
+            &self,
+            _connection: &mut connection::Connection,
+            session_ticket: SessionTicket,
+        ) {
+            let data = session_ticket.session_data();
+            (*self.stored_ticket).borrow_mut().extend(data);
+            // Default ticket lifetime is 15 hours
+            assert_eq!(
+                session_ticket.session_lifetime().unwrap(),
+                Duration::new(54000, 0)
+            );
+        }
+    }
+
+    #[derive(Default, Clone)]
+    pub struct SessionTicketHandler {
+        stored_ticket: Rc<RefCell<Option<SessionTicket>>>,
+    }
+
+    // Implement the session ticket callback that stores the SessionTicket type instead of
+    // raw bytes.
+    impl SessionTicketCallback for SessionTicketHandler {
+        fn on_session_ticket(
+            &self,
+            _connection: &mut connection::Connection,
+            session_ticket: SessionTicket,
+        ) {
+            let mut ptr = (*self.stored_ticket).borrow_mut();
+            if ptr.is_none() {
+                *ptr = Some(session_ticket);
+            }
+        }
+    }
+
+    #[test]
+    fn resume_session() {
```

Comment:
```suggestion
    fn resume_session() -> Result<(), Box<dyn Error>>{
```
If the test returns an Error then you could use the `?` operator to make the config building a little neater.